### PR TITLE
Task 8

### PR DESCRIPTION
**Pull request by jordaninthewind**
This adds backend interaction updates, auth token, observables for loading with httpclient with pagination and filtering.

### CODE REVIEW

File: Angular/angular-mentoring/src/app/course-item/course-item.component.ts [hunk 1]
```diff
@@ -28,8 +28,11 @@ export class CourseItemComponent implements OnInit {
   }
 
   private _selectTileColor() {
-    const isInFuture = this.item.creationDate > new Date();
-    const isNewerThanAFortnite = (new Date().getTime() - this.item.creationDate.getTime()) < this._calculateDays(14);
+    if (this.item === null) return;
+    
+    const date = new Date(`${this.item.date}`);
```
Comment (aledpardo) [inline Angular/angular-mentoring/src/app/course-item/course-item.component.ts:33]: We could just pass `const date = new Date(this.item.date);` right?

File: Angular/angular-mentoring/src/app/services/auth.service.ts [hunk 2]
```diff
@@ -6,42 +7,65 @@ import { UserModel } from '../user-model';
...
+          // this.error = err.error;
+        }
+      )
     return this.user;
   }
 }
+
+const authKey = 'token';
+const authUrl = 'http://localhost:3004/auth/login';
+const userInfoUrl = 'http://localhost:3004/auth/userInfo';
```
Comment (aledpardo) [inline Angular/angular-mentoring/src/app/services/auth.service.ts:71]: Usually we put declarations like this before the class, at the beginning of the file.

Comment (jordaninthewind) [reply Angular/angular-mentoring/src/app/services/auth.service.ts:71]: Good point! Noted!

File: Angular/angular-mentoring/src/app/course-item/course-item.component.ts (hunk 1 above)
Comment (jordaninthewind) [reply Angular/angular-mentoring/src/app/course-item/course-item.component.ts:33]: Apparently so, I swear that it yelled at me for a string type when I first tried to pass in the value. Updated.

**Pull request merged**